kratos/geometries/hexahedra_3d_20.h

```diff
@@ -459,18 +459,32 @@ template<class TPointType> class Hexahedra3D20 : public Geometry<TPointType>
         return Volume(); 
     }
    
-
     /**
-     * Returns whether given arbitrary point is inside the Geometry
+     * Returns whether given arbitrary point is inside the Geometry and the respective 
+     * local point for the given global point
+     * @param rPoint: The point to be checked if is inside o note in global coordinates
+     * @param rResult: The local coordinates of the point
+     * @param Tolerance: The  tolerance that will be considered to check if the point is inside or not
+     * @return True if the point is inside, false otherwise
      */
-    virtual bool IsInside( const CoordinatesArrayType& rPoint, CoordinatesArrayType& rResult, const double Tolerance = std::numeric_limits<double>::epsilon() ) override
+    virtual bool IsInside( 
+        const CoordinatesArrayType& rPoint, 
+        CoordinatesArrayType& rResult, 
+        const double Tolerance = std::numeric_limits<double>::epsilon() 
+        ) override
     {
         this->PointLocalCoordinates( rResult, rPoint );
 
-        if ( fabs( rResult[0] ) <= (1.0 + Tolerance) )
-            if ( fabs( rResult[1] ) <= (1.0 + Tolerance) )
-                if ( fabs( rResult[2] ) <= (1.0 + Tolerance) )
+        if ( std::abs( rResult[0] ) <= (1.0 + Tolerance) )
```

Comment:
i believe we have a problem here, (caming from the past, but hurting here)

my point is that we are using "CoordinatesArrayType" everywhere, that is

     typedef typename PointType::CoordinatesArrayType CoordinatesArrayType;

or in other words
    array_1d<double,3>

the point is that for example for a Triangle3D3, the rResult should be a vector with 2 components (not 3!!)